### DISCUSSION TITLE
[Projects Beta] Add OR, AND, NOT to filter

### DISCUSSION THREAD

**Discussion by mobile-sergey** [community/community#7643] [Projects and Issues] [answered]
I can filter one repo
<img width="512" alt="Снимок экрана 2021-11-12 в 18 43 17" src="https://user-images.githubusercontent.com/1146694/141494020-ca142c05-1e74-49d9-88b8-28c860e0d795.png">

But I need to filter items in project with two repo and I can't do it
<img width="783" alt="Снимок экрана 2021-11-12 в 18 43 02" src="https://user-images.githubusercontent.com/1146694/141494003-b925283b-b327-45c2-91e9-b07ba65a559e.png">

**Comment by maxandersen** [comment, marked as answer]
use a comma:

repo:"repoA","repoB"

that gives your "OR"

A Minus gives you "NOT":

-repo:"repoA"

its not fully generic but it  gets close.

**Reply by mobile-sergey** [reply]
Thanks, your answer fix some cases (and fix case on screenshot).

**Reply by mobile-sergey** [reply]
But not fix complex cases

**Reply by sasklacz** [reply]
How it's supposed to work ? Doesn't look like a valid filter for me

<img width="452" alt="Screenshot 2023-03-14 at 16 42 01" src="https://user-images.githubusercontent.com/112862936/225056050-94c91d2e-6c02-4d62-bca3-d487b58ffbda.png">

**Reply by danielgomezrico** [reply]
That works if you want to filter by one field, but what about filtering by multiple fields?

this does not work
`repo:"repoa",type:""`

**Reply by yenda** [reply]
that does not work as OR for multiple field, this is really basic

**Reply by Thiamath** [reply]
I agree, the `comma` is more like an `IN` clause rather than `OR`.
I can't do `sprint:@previous,<@previous`

Edit: Just realised `sprint:@previous,<@previous` === `sprint:<@current`... 😅 

Some better examples would be `sprint:@next,<@previous` or `no:sprint OR sprint:<@current`

**Comment by truburt** [comment]
One other use-case where OR could be really helpful is to see only issues that miss one of the fields. Imagine I have 'Status', 'Type', 'Date' fields and want to setup a view which shows only the issues that have any of that fields missing.

Filter would look like `no:type OR no:date`. Can't see a way of making this kind of filters atm.

**Reply by mobile-sergey** [reply]
Yes, this case I don't know how to do...

**Reply by leandro-lucarella-frequenz** [reply]
Same for `iteration:>=@current OR no:iteration`. Would be useful to have that.

**Reply by leandro-lucarella-frequenz** [reply]
Alright, the above can be still hacked around using `-iteration:<@current-1`.

**Reply by schweigert** [reply]
for enum, this doesn't work @leandro-lucarella-frequenz

**Comment by salob** [comment]
I don't think this is really resolved, that's fine if you want to search for multiple values for the same field but how would I do the following:

status = "In Progress" OR sprint = "Sprint13"

**Reply by KeeonTabrizi** [reply]
agreed this would make some use-cases much easier - I'm current considering github actions/workflows to overcome this limitation

**Reply by katexochen** [reply]
Yes, we need this.

**Comment by jjstafford** [comment]
This currently works for issues, PLEASE just add the same functionality to projects 🙏🏼 

![image](https://github.com/user-attachments/assets/05633f08-0faf-4f2e-9673-a6218a1e5554)

**Reply by jjstafford** [reply]
This is especially needed since GH just introduced a "Bug" issue type, and we have tons of legacy issues using label "Bug" - this is very hard to manage.
